waspc/data/Generator/templates/react-app/src/auth/useAuth.js

```diff
@@ -11,7 +11,7 @@ async function getMe() {
 
     return response.data
   } catch (error) {
-    if (error.response?.status === 403) {
+    if (error.response?.status === 401 || error.response?.status === 403) {
```

Comment:
I think in this case we probably do, since the auth middleware checks various conditions and returns 401 in the case of some errors: https://github.com/wasp-lang/wasp/blob/811fe57d63d09c92e7bd377d04937477ed443cae/waspc/data/Generator/templates/server/src/core/auth.js We just haven't hit them yet, since they are bad JWT or missing user related.

The main check that fails with 403 is the `/me` one (https://github.com/wasp-lang/wasp/blob/811fe57d63d09c92e7bd377d04937477ed443cae/waspc/data/Generator/templates/server/src/routes/auth/me.js). But that call could in theory return 401 via the middleware.

Reply:
I know see we have an issue for this (I made it and forgot about it): #575

In short, we decided we should change the 403 code to 401 anyway (it's the "correct" one to use). I'm guessing this would remove the need to check for both of them

Reply:
Ok sounds good, thanks for finding that. I can then check for 401 only then and make that change to `/me` as well. 👍🏻